### DISCUSSION TITLE
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE

### DISCUSSION THREAD

**Discussion by EliezerRamirezRuiz** [sqlalchemy/sqlalchemy#11363] [Usage Questions] [answered]
Hello everyone, I'm having a problem when I try to install SQLAlchemy via pip: 

```cmd
(venv) PS C:\Users\eramirez\Desktop\Datavisualizer-website> pip install --upgrade SQLAlchemy
Requirement already satisfied: SQLAlchemy in c:\users\eramirez\desktop\datavisualizer-website\venv\lib\site-packages (2.0.29)
Collecting SQLAlchemy
  Using cached SQLAlchemy-2.0.30-cp312-cp312-win_amd64.whl.metadata (9.8 kB)
Requirement already satisfied: typing-extensions>=4.6.0 in c:\users\eramirez\desktop\datavisualizer-website\venv\lib\site-packages (from SQLAlchemy) (4.10.0)
Requirement already satisfied: greenlet!=0.4.17 in c:\users\eramirez\desktop\datavisualizer-website\venv\lib\site-packages (from SQLAlchemy) (3.0.3)
Downloading SQLAlchemy-2.0.30-cp312-cp312-win_amd64.whl (2.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/2.1 MB ? eta -:--:--
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    unknown package:
        Expected sha256 b6c7ec2b1f4969fc19b65b7059ed00497e25f54069407a8701091beb69e591a5
             Got        e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

¿can Someone help me?. 
I don't know if is my computer, Pypi, pip or SQLAlchemy.

**Comment by gordthompson** [comment]
You might try

```
pip install --no-cache-dir --upgrade SQLAlchemy
```

and see if that helps. More suggestions here:

https://stackoverflow.com/q/40183108/2144390

**Reply by EliezerRamirezRuiz** [reply]
thank you very much friend but in my case it does not work, I feel that the problem comes from somewhere else then I went to see cortex firewall and there comes the possible problem because cortex detects as a malware and I do not know finally if my computer or answer ... 

Just I feel confused, thanks for your time.

**Reply by gordthompson** [reply]
Will your firewall let you download the wheel file directly from here?

https://files.pythonhosted.org/packages/dc/01/bff536f96ea323a7d80df128a7bc947e3c25a60383425bf491232112c30d/SQLAlchemy-2.0.30-cp312-cp312-win_amd64.whl

**Reply by EliezerRamirezRuiz** [reply]
it doesn't work, I have tried to install other libraries or dependencies and they don't throw error but only with SQLAlchemy this happens. I have tried the file you send me and it downloads but its size is 0KB. 

But thank you anyway, have a great time.

![image](https://github.com/sqlalchemy/sqlalchemy/assets/83837915/00aaaa1f-818d-4a4a-91d9-96e194660408)

**Reply by CaselIT** [reply, marked as answer]
the issue seems connection on your side. I can download the same file and it does have the correct side (2.028kb)

**Reply by EliezerRamirezRuiz** [reply]
Thanks to both of you for your help, indeed in my company they installed new certificates that caused the error. I am sorry for wasting your time and thank you again for providing the best availability.
